framework/keyring-interface.md

```diff
@@ -36,37 +36,48 @@ in this document are to be interpreted as described in [RFC 2119](https://tools.
 
 ### OnEncrypt
 
-This interface takes [encryption materials](#structures.md#encryption-materials) as input
-and MAY modify it with any of the following behaviors:
+The following inputs are REQUIRED:
+
+- [Algorithm Suite](#algorithm-suites.md)
+- [Encryption Context](#encryption-context)
+
+The following inputs are OPTIONAL:
+
+- [Plaintext Data Key](#structures.md#plaintext-data-key)
+
+This interface MAY return [data key materials](#structures.md#data-key-materials) appropriate for the request. If it does return this output
+and a plaintext data key was provided as input, the output must specify the same plaintext data key.
```

Comment:
`on encrypt` MUST always emit data key materials. They MAY have additional contents that were not present in the input materials.


Changing to "return nothing means nothing changed" is a fundamental change to the keyring interface and seems to place the onus of consolidating materials on the caller rather than the keyring.